src/main/java/seedu/address/model/person/SalaryRange.java

```diff
@@ -0,0 +1,175 @@
+            + "4294967295] "
+            + "or two pure digital numbers with ‘-’ in between or '<=' or '>=' followed by a digital number.\n"
+            + "Numbers can vary from large to small or from small to large.\n"
+            + "Both digital numbers should be within the range [0, 2147483647]\n"
+            + "Examples:\n"
+            + "2000-4000\n"
+            + ">=5000\n"
+            + "<=10000\n";
+    public static final String VALIDATION_REGEX = "^(\\d+-\\d+)|((<=|>=)\\d+)|(\\d+)$";
+    public static final int UPPERBOUND = 2147483647;
+    public static final int LOWERBOUND = 0;
+
+    private int maxSalary;
+    private int minSalary;
+    private boolean isRange;
+    private boolean isMax;
+    private boolean isMin;
+
+    /**
+     * Constructs a {@code Salary}.
+     *
+     * @param salaryRange A valid salary range.
+     */
+    public SalaryRange(String salaryRange) {
+        requireNonNull(salaryRange);
+        checkArgument(isValidSalaryRange(salaryRange), MESSAGE_CONSTRAINTS);
+        parseSalaryRange(salaryRange);
+    }
+
+    /**
+     * Constructs a {@code Salary}.
+     *
+     * @param salaryRange A valid salary in string format.
+     */
+    public void parseSalaryRange(String salaryRange) {
+        if (salaryRange.contains("-")) {
+            String[] range = salaryRange.split("-");
+            int minSalary = Integer.parseInt(range[0]);
+            int maxSalary = Integer.parseInt(range[1]);
+            this.minSalary = Math.min(minSalary, maxSalary);
+            this.maxSalary = Math.max(minSalary, maxSalary);
+            this.isRange = true;
+        } else if (salaryRange.contains(">=")) {
+            String[] range = salaryRange.split(">=");
+            this.minSalary = Integer.parseInt(range[1]);
+            this.maxSalary = UPPERBOUND;
+            this.isMin = true;
+        } else if (salaryRange.contains("<=")) {
+            String[] range = salaryRange.split("<=");
+            this.maxSalary = Integer.parseInt(range[1]);
+            this.minSalary = LOWERBOUND;
+            this.isMax = true;
+        } else {
+            this.minSalary = Integer.parseInt(salaryRange);
+            this.maxSalary = Integer.parseInt(salaryRange);
+            this.isRange = false;
+        }
+    }
+
+    /**
+     * Returns true if a given string is a valid salary format.
+     */
+    public static boolean isValidSalaryRange(String test) {
+        if (!test.matches(VALIDATION_REGEX)) {
+            return false;
+        }
+        if (test.contains("-")) {
+            return isValidRange(test);
+        } else if (test.contains(">=")) {
+            return isValidMin(test);
+        } else if (test.contains("<=")) {
+            return isValidMax(test);
+        } else {
+            return isValidNum(test);
+        }
+    }
+    private static boolean isValidRange(String test) {
```

Comment:
Can try to ensure that all methods are being tested for to increase code coverage